src/libtomahawk/accounts/DelegateConfigWrapper.cpp

```diff
@@ -227,8 +227,30 @@ DelegateConfigWrapper::onConfigTestResult( Tomahawk::Accounts::ConfigTestResultT
 
         m_invalidData = m_widget->readData();
 
-        // TODO: generate message based on status code
-        m_errorLabel->setText( QString( "<font color='red'>%1</font>" ).arg( tr( "Your config is invalid." ) ) );
+	QString msg;
+	if (message.isEmpty()){
+	    msg = getTestConfigMessage(code);
+	} else {
+	    msg = message;
+	}
+        m_errorLabel->setText( QString( "<font color='red'>%1</font>" ).arg( msg ) );
+    }
+}
+
+QString
+DelegateConfigWrapper::getTestConfigMessage( int code )
+{
+    switch(code) {
+	case Tomahawk::Accounts::ConfigTestResultCommunicationError:
+	    return QObject::tr( "Unable to authenticate. Please check your connection." );
+	case Tomahawk::Accounts::ConfigTestResultInvalidCredentials:
+	    return QObject::tr( "Username or password incorrect." );
+	case Tomahawk::Accounts::ConfigTestResultInvalidAccount:
+	    return QObject::tr( "Account rejected by server." );
+	case Tomahawk::Accounts::ConfigTestResultPlayingElsewhere:
+	    return QObject::tr( "Action not allowed, account is in use elsewhere." );
+	case Tomahawk::Accounts::ConfigTestResultAccountExpired:
+	    return QObject::tr( "Your account has expired." );
```

Comment:
You can probably just use `tr()` instead of `Object::tr()` here, because DelegateConfigWrapper is most probably a `QObject`, it does not harm though ;-)